scripts/test-cli.sh

```diff
@@ -48,11 +48,27 @@ test -s out || fail "scalajsp bin/Foo$.sjsir: empty output"
 scalajsp bin/Foo\$A.sjsir > out
 test -s out || fail "scalajsp bin/Foo\$A.sjsir: empty output"
 
-scalajsld -s -o test.js -mm Foo.main bin
+scalajsld -s -o test.js -mm Foo.main bin 2> test_stderr.txt
+grep -Fxq "Warning: using file as output is deprecated since 1.3.0. Please provide directory" test_stderr.txt \
+  || fail "expected warning. Got: $(cat test_stderr.txt)"
 test -s test.js || fail "scalajsld: empty output"
 test -s test.js.map || fail "scalajsld: empty source map"
 
-node test.js > got.run
+node test.js > got-legacy.run
+cat > want-legacy.run <<EOF
+asdf 2
+EOF
+
+diff got-legacy.run want-legacy.run
+
+mkdir test-output
+scalajsld -s --outputDir test-output --moduleSplitStyle SmallestModules --moduleKind CommonJSModule -mm Foo.main bin
+test -s test-output/Foo.js || fail "scalajsld: empty output"
+test -s test-output/Foo.js.map || fail "scalajsld: empty output"
+test -s test-output/main.js || fail "scalajsld: empty output"
+test -s test-output/main.js.map || fail "scalajsld: empty source map"
```

Comment:
It's maybe a good idea to have more specific error messages here. Otherwise it's hard to understand which part of the output is empty. (n.b. `fail` will not report the line it failed on, so calling it twice with the same string should be avoided if possible).